rasa/nlu/data_augmentation/augmenter.py

```diff
@@ -0,0 +1,638 @@
+
+    Args:
+         output_directory: Directory to store the evaluation reports and augmented training data in.
+         nlu_training_file: Augmented NLU training data file.
+         config: NLU model config.
+         nlu_evaluation_data: NLU evaluation data.
+
+    Returns:
+        Classification report of the NLU model trained on the augmented training data.
+    """
+
+    # Train NLU model
+    model_path = train_nlu(
+        config=config, nlu_data=nlu_training_file, output=output_directory
+    )
+
+    # Load and Evaluate NLU Model
+    unpacked_model_path = get_model(model_path)
+    nlu_model_path = os.path.join(unpacked_model_path, "nlu")
+
+    interpreter = Interpreter.load(nlu_model_path)
+    interpreter.pipeline = remove_pretrained_extractors(interpreter.pipeline)
+
+    (intent_results, *_) = get_eval_data(interpreter, nlu_evaluation_data)
+    intent_report = create_intent_report(
+        intent_results=intent_results, output_directory=None, report_as_dict=True,
+    )
+    intent_errors = get_intent_errors(intent_results=intent_results)
+    rasa.shared.utils.io.dump_obj_as_json_to_file(
+        os.path.join(output_directory, "intent_errors.json"), intent_errors,
+    )
+
+    return intent_report["report"]
+
+
+def _create_augmentation_summary(
+    intents_to_augment: Set[Text],
+    changed_intents: Set[Text],
+    classification_report: Dict[Text, Dict[Text, float]],
+    intent_report: Dict[Text, float],
+) -> Tuple[
+    Dict[Text, Dict[Text, float]], Dict[Text, float],
+]:
+    """Creates a summary report of the effect of data augmentation and modifies the original classification report
+    with that information.
+
+    Args:
+        intents_to_augment: The intents that have been selected for data augmentation.
+        changed_intents: The intents that have been affected by data augmentation.
+        classification_report: Classification report of the model run *without* data augmentation.
+        intent_report: Report of the model run *with* data augmentation.
+
+    Returns:
+        A tuple representing a summary of the changed intents as well as a modified version of the original classification report with performance changes for all affected intents.
+    """
+
+    intent_summary = collections.defaultdict(dict)
+
+    # accuracy is the only non-dict like thing in the classification report, so it receives extra treatment
+    if "accuracy" in classification_report:
+        accuracy_change = intent_report["accuracy"] - classification_report["accuracy"]
+        acc_dict = {
+            "accuracy_change": accuracy_change,
+            "accuracy": intent_report["accuracy"],
+        }
+
+        intent_report["accuracy"] = acc_dict
+        intent_summary["accuracy"] = {"accuracy_change": accuracy_change}
+
+    for intent in (
+        intents_to_augment
+        | changed_intents
+        | {"micro avg", "macro avg", "weighted avg"}
+    ):
```

Comment:
For more clarity, we could store the value of `intents_to_augment | changed_intents | {"micro avg", "macro avg", "weighted avg"}` in a variable, and then use it in the `for` loop.